pkg/cloud/baremetal/actuators/machine/actuator.go

```diff
@@ -430,6 +438,164 @@ func consumerRefMatches(consumer *corev1.ObjectReference, machine *machinev1.Mac
 	return true
 }
 
+// updateUserData fetches full ignition from MCS and updates the UserData secret reference
+func (a *Actuator) updateUserData(ctx context.Context, config *bmv1alpha1.BareMetalMachineProviderSpec,
+	machine *machinev1.Machine) error {
+
+	// Try to fetch the full ignition secret to see if it exists
+	machineUserDataSecret := &corev1.Secret{}
+	key := client.ObjectKey{
+		Name:      machine.Name + "-user-data",
+		Namespace: machine.Namespace,
+	}
+	// Check if the full ignition secret already exists. If not, create one
+	err := a.client.Get(ctx, key, machineUserDataSecret)
+	if errors.IsNotFound(err) {
```

Comment:
Just for readability, I'd probably turn this around as sketched below. But there's nothing wrong with how it is now.

```
if err == nil {
  return nil
}
if !errors.IsNotFound(err) {
  // log error
  return err
}

// now proceed with the rest of the code without the indentation and uncertainty
// about what comes at the end. That way we know clearly from here on down
// what the goal is.
```

Reply:
Doh!! I'm running around like a headless chicken. This is so much easier to read.